### PR TITLE
Update all the requirements

### LINKED ISSUE #100: Add a support for xor ranges, when they are introduced to yaramod
**Feature Category**

- [x] Correctness
- [ ] User Interface / User Experience
- [ ] Performance
- [ ] Other (please explain)

**Describe the problem**

We don't support yara xor modifiers (only xor with range 0-255). This is currently not exposed by upstream.

**Describe the solution you'd like**

Fix this in upstream or ask them to do this, and later use this in our code.

### LINKED ISSUE #231: Fix regex parsing (in avast/yaramod)
Discovered during testing a PR.

There is a bug in upstream regex parsing. Already reported here: https://github.com/avast/yaramod/issues/186. We should help yaramod fix this issue (or wait for when they have time).

### LINKED ISSUE #235: Bump dependencies to the current version
**Feature Category**

- [ ] Correctness
- [ ] User Interface / User Experience
- [ ] Performance
- [x] Other (please explain) - Maintenance

**Describe the problem**

We pin all of our dependencies.
Because of that, some of our dependencies are outdated

**Describe the solution you'd like**

Bump all the dependencies to the newest possible version. Test the system after that.

**Describe alternatives you've considered**

"If it's not broken, don't fix it" - maybe, but there are probably hiden bugs that we didn't notice yet. Especially in the UI code.

### PR DESCRIPTION
**Pull request by msm-code**
**Your checklist for this pull request**
- [x] I've read the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [x] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)
- [ ] I've added automated tests for my change (if applicable, optional)
- [ ] I've updated documentation to reflect my change (if applicable)

**What is the current behaviour?**
Dependencies are updated

**What is the new behaviour?**
Suprisingly, everything seems to work after updating the dependenceis (and minor fixes). I'll do more testing later, but I think tihs is ready for review.

**Test plan**
Click around

**Closing issues**

fixes #231 (tested)
fixes #235
fixes #100

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -63,7 +62,7 @@ def download(job_id: str, ordinal: int, file_path: str) -> FileResponse:
     arbitrary files (for example "/etc/passwd").
     """
     if not db.job_contains(JobId(job_id), ordinal, file_path):
-        raise NotFound("No such file in result set.")
+        raise Response("No such file in result set.", status_code=404)
```
Comment (nazywam) [inline src/app.py:65]:
```suggestion
        return Response("No such file in result set.", status_code=404)
```
probably?

**Review by nazywam — CHANGES_REQUESTED** [review]
Looks good, the other failed tests seem unrelated.

What's the usecase of `requirements.plain.txt`, should we maybe put it in a comment somewhere?

**Comment by msm-code** [comment]
Uh oh, thanks for catching this. I've actually fixed this and forgot to commit :facepalm:.

I'll apply the changes and do some more testing.

**Comment by msm-code** [comment]
> What's the usecase of requirements.plain.txt, should we maybe put it in a comment somewhere?

The idea was that I had to do some guesswork to find "original" dependencies. As you can see we had some "old" dependencies that were removed (for example, werkzeug), but stayed in the requirements because they were a transitive dependency of some project.

Basically the difference between packages.json and packeges.lock or something. I don't know a good place for a file like this - any ideas?

**Comment by msm-code** [comment]
Actually this also fixes #100 - added to the description.

**Review by nazywam — APPROVED** [review]

**Pull request merged**